openweather-based-bidding/main.gs

```diff
@@ -26,97 +26,66 @@ const sheetsApi = new SheetsApi(configSpreadsheetId);
   const auth     = new Auth(config.get('service-account'));
   const dv360    = new DV360(auth.getAuthToken());
   const weather  = new OpenWeather(config.get('open-weather-api-key'));
 
-  // Get config from the spreadsheet
-  const sheet = SpreadsheetApp.openById(configSpreadsheetId)
-    .getSheetByName(configSpreadsheetName);
-  if (!sheet) {
-    throw 'Cannot find spreadsheet with the name: ' + configSpreadsheetName;
-  }
+  sheetsApi.getSheetObject();
 
   for (let i = 1; i < rows.length; i++) {
     const row = rows[i];
-
-    const lineItemId        = parseInt(row[config.get('col-line-item-id')]),
-        insertionOrderId    = parseInt(row[config.get('col-insertion-order-id')]),
-        advertiserId        = parseInt(row[config.get('col-advertiser-id')]),
-        lat                 = row[config.get('col-lat')],
-        lon                 = row[config.get('col-lon')],
-        tempMin             = row[config.get('col-temp-min')],
-        tempMax             = row[config.get('col-temp-max')],
-        onlyIfPrecipitation = row[config.get('col-precip')] === 'TRUE',
-        windMin             = row[config.get('col-wind-min')];
-
-    // Get current weather conditions
-    const currentWeather = weather.getCurrent(lat, lon);
-
-    // Update row with current weather data
-    row[config.get('col-temp-curr')] = currentWeather.temp;
-    row[config.get('col-precip-curr')] = currentWeather.hasOwnProperty('snow')
-      || currentWeather.hasOwnProperty('rain');
-    row[config.get('col-wind-curr')] = currentWeather.wind_speed;
-
-    // Initialize all conditions as satisfied
-    const satisfiedConditions = {
-      tempMin: true,
-      tempMax: true,
-      precip: true,
-      windSpeed: true,
-    };
-
-    // Check if min temperature condition is satisfied
-    if (tempMin && !isNaN(tempMin)) {
-      satisfiedConditions.tempMin = parseFloat(tempMin) <= parseFloat(row[config.get('col-temp-curr')]);
-    }
-
-    // Check if max temperature condition is satisfied
-    if (tempMax && !isNaN(tempMax)) {
-      satisfiedConditions.tempMax = parseFloat(row[config.get('col-temp-curr')]) <= parseFloat(tempMax);
-    }
-
-    // Check if precipitation condition is satisfied
-    if (onlyIfPrecipitation) {
-      satisfiedConditions.precip = row[config.get('col-precip-curr')];
+    const iPlus1 = i + 1;
+
+    const lineItemId = parseInt(row[config.getHeaderIndex('col-line-item-id')]),
+          insertionOrderId = parseInt(
+            row[config.getHeaderIndex('col-insertion-order-id')]
+          ),
+          advertiserId = parseInt(
+            row[config.getHeaderIndex('col-advertiser-id')]
+          ),
+          lat = parseFloat(row[config.getHeaderIndex('col-lat')]),
+          lon = parseFloat(row[config.getHeaderIndex('col-lon')]);
+
+    // Get weather conditions
+    const allWeather = weather.getAll(lat, lon);
+
+    // Extract all weather variables and write their values to the spreadsheet
```

Comment:
We're not writing to the spreadsheet at this point.
`// Extract weather conditions` may be sufficient?

Reply:
Good one, thanks